packages/docs/src/stories/templates/header.stories.ts

```diff
@@ -276,11 +308,25 @@ export const SampleB = {
         </sd-navigation-item>
       </nav>
     </sd-drawer>
-    <script>
-      function openDrawerSampleB() {
+    <script type="module">
+      await Promise.all([
+        customElements.whenDefined('sd-navigation-item'),
+        customElements.whenDefined('sd-drawer')
+      ]).then(() => {
         const drawerSampleB = document.getElementById('sample-b-drawer');
-        drawerSampleB.show();
-      }
+        const navigationItemSampleB = document.getElementById('open-menu-sample-b');
+        const buttonSampleB = navigationItemSampleB.shadowRoot.querySelector('button');
+
+        buttonSampleB.setAttribute('aria-controls', 'sample-b-drawer');
+        //Add the necessary ARIA attributes to prevent only being added after action
+        buttonSampleB.setAttribute('aria-expanded', 'false');
+        drawerSampleB.addEventListener('sd-hide', () => buttonSampleB.setAttribute('aria-expanded', 'false'));
+
+        navigationItemSampleB.addEventListener('click', () => {
+          drawerSampleB.show();
+          buttonSampleB.setAttribute('aria-expanded', 'true');
```

Comment:
And here.

Reply:
Same here https://github.com/solid-design-system/solid/pull/1668#discussion_r1863120614